commands/src/api/camera.rs

```diff
@@ -0,0 +1,57 @@
+pub use lumeo_events::camera::Camera;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use url::Url;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum Request {
+    /// Discover cameras. Returned data can be incomplete if camera
+    /// requires authorization. For authorized access use `GetWithAuth`.
+    Discover,
+
+    /// Get camera information
+    GetWithAuth {
+        url: Url,
+        credentials: Option<Credentials>,
+    },
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct DiscoverResponse(pub Vec<Camera>);
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct GetWithAuthResponse(pub Result<Camera, GetWithAuthError>);
+
+#[derive(Serialize, Deserialize, Debug, Clone, Error)]
+pub enum GetWithAuthError {
+    #[error("Camera is offline")]
+    CameraOffline,
+
+    #[error("Invalid credentials")]
+    InvalidCredentials,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Credentials {
+    /// Camera user name
+    pub username: String,
+
+    /// Plaintext password
+    pub password: String,
+}
+
+/// Serializable variant of std::result::Result
```

Comment:
`std::result::Result` can be deserialized ([docs](https://docs.rs/serde/1.0.115/serde/trait.Deserialize.html#impl-Deserialize%3C%27de%3E-for-Result%3CT%2C%20E%3E)).

Reply:
Thanks!